src/SourceBuild/content/Directory.Build.targets

```diff
@@ -26,4 +26,22 @@
     </PropertyGroup>
   </Target>
 
-</Project>
\ No newline at end of file
+  <!-- Create a merge manifest from the individual repository manifest files. -->
+  <UsingTask TaskName="Microsoft.DotNet.UnifiedBuild.Tasks.MergeAssetManifests" AssemblyFile="$(MicrosoftDotNetUnifiedBuildTasksAssembly)" TaskFactory="TaskHostFactory" />
+  <Target Name="MergeAssetManifests" AfterTargets="Build">
+    <PropertyGroup>
+      <MergedAssetManifestOutputPath>$(ArtifactsDir)VerticalManifest.xml</MergedAssetManifestOutputPath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <RepoAssetManifest Include="$(AssetManifestsIntermediateDir)\**\*.xml" />
+    </ItemGroup>
+
+    <!-- It's OK for the VmrBuildNumber to be empty -->
+    <Microsoft.DotNet.UnifiedBuild.Tasks.MergeAssetManifests
+      AssetManifest="@(RepoAssetManifest)"
+      MergedAssetManifestOutputPath="$(MergedAssetManifestOutputPath)"
+      VmrBuildNumber="$(BUILD_BUILDNUMBER)" />
+  </Target>
```

Comment:
This looks wrong. In latest main, publish.proj handles this. You just need to add a P2P in your test application to publish.proj to make sure this runs early enough. Something like:

```
<ItemGroup>
  <ProjectReference Include="$(RepositoryEngineeringDir)publish.proj" ReferenceOutputAssembly="false" PrivateAssets="all" />
</ItemGroup>
```

Reply:
Thanks for the heads up, I hadn't merged main in a while, that's a much nicer solution.